src/lib/services/api-token-service.ts

```diff
@@ -46,15 +47,25 @@ export class ApiTokenService {
 
     private timer: NodeJS.Timeout;
 
+    private seenTimer: NodeJS.Timeout;
+
     private activeTokens: IApiToken[] = [];
 
+    private lastSeenSecrets: string[] = [];
```

Comment:
One problem with this approach could be memory consumption, cause we might end up with a large array in memory. I think it'd be better to have a Map structure and then use the keys to obtain the last seen secrets.

Reply:
The array gets cleared in https://github.com/Unleash/unleash/pull/2514/files/e31c6d6296e7b5317e86e502abbbe4ff158c0b76#diff-c7747d86059ec3422885e79411b8ef7323c0804df081c110268dd3250ace3696R102 - But we can also try a Map approach.

Reply:
I remembered that you may be thinking about having an unique array (don't have the same token more than once in the array). I agree with that and I think we could maybe use a Set here.

Reply:
Yes, a set would work as well, the problem is with high-throughput applications, say 1000 req/s we'd be queuing 60K strings in memory every minute (which is 4.2Mb considering a token size of 70 characters). It's not much but if we're speaking of a single token, it'd be much better to have only 70 bytes in memory :D

Reply:
Thanks, tried addressing this in https://github.com/Unleash/unleash/pull/2514/commits/ed33001b04d69c056ed25827d2530a777e88ecf1 - Let me know what you think 👍 
I'll do the same change in the PAT PR.